src/sparkx/Jetscape.py

```diff
@@ -222,7 +222,32 @@ def _particle_as_list(self, particle: Particle) -> List[Union[int, float]]:
         particle_list[6] = float(particle.pz)
 
         return particle_list
+    
+    def _update_after_merge(self, other: BaseStorer) -> None:
+        """
+        Updates the current instance after merging with another Jetscape instance.
+
+        This method is called after merging two Jetscape instances to update the
+        attributes of the current instance based on the attributes of the other instance.
+        The last line, filename and :code:`sigmaGen` are taken from the left-hand instance.
+
+        Parameters
+        ----------
+        other : Jetscape
+            The other Jetscape instance that was merged with the current instance.
 
+        Raises
+        ------
+        UserWarning
+            If the Jetscape :code:`particle_type_` or :code:`particle_type_defining_string_` of the two instances do not match, a warning is issued.
+        """
+        if not isinstance(other, Jetscape):
+            raise TypeError("Can only add Jetscape objects to Jetscape.")
+        if self.particle_type_ != other.particle_type_:
+            warnings.warn("particle_types of the merged instances do not match. Taking the left-hand side particle type.")
```

Comment:
Are you sure you want to make this only a warning? This would mean one can mix up partons and hadrons by adding two different Jetscape objects. I would probably go for an error here.

Reply:
Okay in the case of Jetscape I see the issue. Different formats would probably be phyiscs-wise incompatible.

Reply:
Adapted.